syntax = "proto3";

package search;

// Definicion del servicio de busqueda
service ServicioBusqueda
{
    rpc ObtenerRespuesta(Nombre) returns (ListaResultante) {}
}

// Definicion del nombre del producto
message Nombre
{
    string nombre = 1;
}

// Definicion de la lista que se retorna al cliente
message ListaResultante
{
    repeated Item item = 1;
}

// Definicion de los atributos del producto
message Item
{
    int32 id = 1;
    string nombre = 2;
    string marca = 3;
    int32 precio = 4;
}